sdk/src/azure/core/az_http_request.c

```diff
@@ -35,8 +30,15 @@ AZ_NODISCARD az_result az_http_request_init(
   _az_PRECONDITION_VALID_SPAN(headers_buffer, 0, false);
 
   int32_t query_start = 0;
-  az_result url_with_query = _az_span_scan_until(
-      az_span_create(az_span_ptr(url), url_length), _az_is_question_mark, &query_start);
+  uint8_t* ptr = az_span_ptr(url);
+  for (; query_start < url_length; ++query_start)
+  {
+    uint8_t next_byte = ptr[query_start];
+    if (next_byte == '?')
```

Comment:
```suggestion
    if (ptr[query_start] == '?')
```

Reply:
I prefer keeping the local to be consistent with other places where `next_byte` is used multiple times. This way, if we need to access the `next_byte` here in a different place, we have it already. There is no code size difference between the two (also slightly better reaability).